node/lib/DeviceAccessor.ts

```diff
@@ -1,18 +1,89 @@
+     * around require(). Throws if the module could not be loaded.
+     *
+     * @param {string} interfaceModuleName  Package-qualified name (or relative path) of
+     *     the interface module. The package-qualified name can be obtained from
+     *     PackageTranslatorInfo.interfaces.
+     *
+     * @returns {Promise<DeviceInterface>} The loaded device interface.
+     */
+    public static getInterfaceAsync(interfaceModuleName: string): Promise<DeviceInterface>;
+
+    /**
+     * Loads a DeviceInterface from a module. This is just a convenience wrapper
+     * around require(). Throws if the module could not be loaded.
+     *
+     * @param {string} packageName  Name (or relative path) of the package containing
+     *     the module. The name can be obtained from PackageInfo.name.
+     * @param {string} interfaceName  Simple name of the interface module. Can be obtained
+     *     from PackageInterfaceInfo.name.
+     * @returns {Promise<DeviceInterface>} The loaded device interface.
+     */
+    public static getInterfaceAsync(
+            packageName: string, interfaceName: string): Promise<DeviceInterface>;
+
+    /**
+     * Loads a DeviceInterface from a module. (Overloaded method implementation.)
+     */
+    public static getInterfaceAsync(
+            packageNameOrInterfaceModuleName: string,
+            interfaceName?: string): Promise<DeviceInterface> {
+        let interfaceModulePath: string = (interfaceName ?
+                packageNameOrInterfaceModuleName + "/" + interfaceName :
+                packageNameOrInterfaceModuleName);
+        return new Promise<DeviceInterface>((resolve, reject) => {
+            let deviceInterface: DeviceInterface;
+            try {
+                deviceInterface = require(interfaceModulePath);
+            } catch (err) {
+                reject(err);
+                return;
+            }
+            resolve(deviceInterface);
+        });
+    }
+
+    /**
+     * Loads a translator from a module. This is just a convenience wrapper
+     * around require(). Throws if the module could not be loaded.
+     *
+     * @param {string} packageName  Name (or relative path) of the package containing
+     *     the module. The name can be obtained from PackageInfo.name.
+     * @param {string} translatorName  Name of the translator module. Can be obtained from
+     *     PackageTranslatorInfo.name.
+     * @param {*} properties  Property bag to be passed to the translator constructor.
+     * @returns {Promise<ITranslator>} The loaded translator instance.
+     */
+    public static async createTranslatorAsync(
+            packageName: string, translatorName: string, properties: any): Promise<ITranslator> {
+        return new Promise<ITranslator>((resolve, reject) => {
+            let translator: ITranslator;
+            try {
+                let translatorClass: any = require(packageName + "/" + translatorName);
```

Comment:
The reason for the separation is they would commonly be obtained separately via `PackageInfo.name` and `PackageTranslatorInfo.name`. But probably I should add an overload that does allow a fully-qualified module name, exactly like I did for `getInterfaceAsync()`.

---
In reply to: [73273305](https://github.com/openT2T/opent2t/pull/2#discussion_r73273305) [](ancestors = 73273305)